lib/msf/core/exploit/smb/mitm_dispatcher.rb

```diff
@@ -0,0 +1,321 @@
+module Msf
+class Exploit
+module SMB
+  # This class provides a Man-In-The-Middle packet Dispatcher.
+  class MitmDispatcher < RubySMB::Dispatcher::Base
+      READ_TIMEOUT = 30
+      ARP_DELAY = 1.5
+
+      # The underlying socket that we select on
+      # @!attribute [rw] tcp_socket
+      #   @return [IO]
+      attr_accessor :tcp_socket
+
+      # The read timeout
+      # @!attribute [rw] read_timeout
+      #   @return [Integer]
+      attr_accessor :read_timeout
+
+      # The arp delay
+      # @!attribute [rw] arp_delay
+      #   @return [Float]
+      attr_accessor :arp_delay
+
+      # The mitm interface
+      # @!attribute [rw] interface
+      #   @return [String]
+      attr_accessor :interface
+
+      # The mac address
+      # @!attribute [rw] mac
+      #   @return [String]
+      attr_accessor :mac
+
+      # The destination mac address
+      # @!attribute [rw] dstmac
+      #   @return [String]
+      attr_accessor :dstmac
+
+      # The sequence number
+      # @!attribute [rw] seqnum
+      #   @return [Integer]
+      attr_accessor :seqnum
+
+      # The acknowledgement number
+      # @!attribute [rw] acknum
+      #   @return [Integer]
+      attr_accessor :acknum
+
+      attr_accessor :ip_src
+
+      attr_accessor :ip_dst
+
+      attr_accessor :tcp_src
+
+      attr_accessor :tcp_dst
+
+      attr_accessor :tcp_win
+
+      # Current Capture
+      attr_accessor :ccap
+
+      # Wether we are still forwarding packets
+      # @!attribute [rw] forwarding
+      #   @return [bool]
+      attr_accessor :forwarding
+
+      attr_accessor :override_client_instance_variables
+      # These are overrides for the client instance variables
+      attr_accessor :dialect
+      attr_accessor :session_id
+      attr_accessor :smb1
+      attr_accessor :smb2
+      attr_accessor :smb3
+      attr_accessor :smb2_message_id
+      attr_accessor :server_guid
+      attr_accessor :negotiated_smb_version
+      attr_accessor :session_encrypt_data
+
+      # We decide the host based on the first person to connect
+      def initialize(interface: PacketFu::Utils.default_int, mac: PacketFu::Utils.whoami?(iface: @interface)[:eth_saddr], read_timeout: READ_TIMEOUT, arp_delay: ARP_DELAY)
+        @override_client_instance_variables = true
+        @interface = interface
+        @mac = mac
+        @read_timeout = read_timeout
+        @arp_delay = arp_delay
+        @smb1 = false
+        @smb2 = true
+        @smb3 = false
+        @session_encrypt_data = false
```

Comment:
These values are not used in this class, just instantiate the client with these values passed as keyword arguments.
```suggestion
```